src/main.rs

```diff
@@ -177,188 +225,225 @@ async fn sync(
+                config.pg_batch_size, 
+                config.blaze_page_resource_count).await?;
+        }
+    } else if blaze_available && !all_tables_exist {
+        create_tables(pg_con_pool).await?;
+        for type_arg in type_args {
+            sync_blaze_2_pg(
+                &config.blaze_base_url,
+                pg_con_pool,
+                type_arg,
+                config.pg_batch_size,
+                config.blaze_page_resource_count).await?;
+        }
+    }
     Ok(())
 }
 
-//@todo: add check whether to number of blaze resources exactly matches the number of resources in pg after inserts
 
 #[tokio::main]
 async fn main() -> Result<(), anyhow::Error>{
-    // Initialize tracing subscriber
+
     tracing_subscriber::fmt()
     .with_max_level(tracing::Level::INFO)
     .init();
 
-    //let args: Vec<String> = env::args().collect();
-    /*
-    if args.len() != 3 || args[1] != "upsert" {
-        eprintln!("Usage: fhir2sql upsert <type>. \n
-        <type> must be one of observation, patient, specimen, or condition");
-        return;
-    }
-    let type_arg = &args[2];
-    // Call your upsert function with the type argument
-    upsert(type_arg);
-    */
-    //@todo: add error handling and stuff
-    let type_arg = "Specimen";
-
     dotenv().ok();
 
-    let blaze_base_url: String = dotenv::var("BLAZE_BASE_URL").expect("BLAZE_BASE_URL must be set");
-    let host = dotenv::var("PG_HOST").expect("PG_HOST must be set");
-    let port = dotenv::var("PG_PORT").expect("PG_PORT must be set");
-    let username = dotenv::var("PG_USERNAME").expect("PG_USERNAME must be set");
-    let password = dotenv::var("PG_PASSWORD").expect("PG_PASSWORD must be set"); //@todo: move out of .env
-    let dbname = dotenv::var("PG_DBNAME").expect("PG_DBNAME must be set");        
-    let batch_size = match dotenv::var("PG_BATCH_SIZE") {
-        Ok(val) => match val.parse::<u32>() {
-            Ok(num) => num,
-            Err(_) => {
-                eprintln!("PG_BATCH_SIZE must be a positive number. Using default value.");
-                10000 // default value if parsing fails
+    //@todo: make use of clap crate? 
+    let config = Config {
+        blaze_base_url: env::var("BLAZE_BASE_URL").expect("BLAZE_BASE_URL must be set"),
+        pg_host: env::var("PG_HOST").expect("PG_HOST must be set"),
+        pg_username: env::var("PG_USERNAME").expect("PG_USERNAME must be set"),
+        pg_password: env::var("PG_PASSWORD").expect("PG_PASSWORD must be set"),
+        pg_dbname: env::var("PG_DBNAME").expect("PG_DBNAME must be set"),
+        pg_port: 5432,
+        pg_batch_size: 10000,
+        blaze_page_resource_count: 5000,
+        blaze_num_connection_attempts: 20,
+        target_time: match NaiveTime::from_hms_opt(3, 0, 0) {
+            Some(time) => time,
+            None => {
+                error!("Invalid target time");
+                return Err(anyhow::Error::msg("Invalid target time"));
             }
```

Comment:
I would use `.ok_or(anyhow::anyhow!("Invalid target time"))?`. The log is not really needed as the program will exit with the error anyway here and the `anyhow!` macro can me used to easily create errors.